miprometheus/problems/question_context_to_class/babiqa_dataset_single_question.py

```diff
@@ -0,0 +1,514 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) IBM Corporation 2018
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, e:wqither express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""bAbiQA.py: contains code for loading the babi dataset (based on the parsing used in torchtext)"""
+__author__= "Vincent Albouy, Ryan.L McAvoy"
+
+import torch
+from miprometheus.utils.problems_utils.language import Language
+import torch.utils.data
+from tqdm import tqdm
+import os
+from miprometheus.utils.app_state import AppState
+from miprometheus.problems.seq_to_seq.seq_to_seq_problem import SeqToSeqProblem
+from miprometheus.utils.loss.masked_cross_entropy_loss import MaskedCrossEntropyLoss
+
+
+
+class BABI(SeqToSeqProblem):
+    """
+    Problem Class for loading bAbi QA data set using Torchtext
+    
+    Inherits from SeqToSeqProblem
+    
+    """
+
+    def __init__(self, params):
+        """
+   
+        Initializes BABI QA problem, calls base class initialization, sets properties using the provided parameters.
+
+        :param params: Dictionary of parameters (read from configuration file).
+        
+        """
+        super(BABI).__init__()
+
+        self.directory = './'
+
+        # boolean: is it training phase?
+        self.data_type = params['data_type']
+
+        self.use_batches = params['batch_size']
+
+        # task number to train on
+        self.tasks = params['tasks']
+
+        self.loss_function = MaskedCrossEntropyLoss()
+
+        self.tenK = params['ten_thousand_examples']
+
+        self.one_hot_embedding = params['one_hot_embedding']
+
+        self.batch_size = params['batch_size']
+
+        self.memory_size = params['truncation_length']
+
+        self.embedding_type = params['embedding_type']
+
+        self.embedding_size = 38
```

Comment:
The size of embedding is hardcoded, whereas, if I understand it correctly, you can use different ones by setting 'embedding type'. Is that ok?

Reply:
yes , I put it in the yaml as parameters but the parser couldn't find it. Will try again.